Изменения компонента maps

ALNovak opened the pull request.

Commit:
Saving files before refreshing line endings

ALNovak committed:
Merge branch 'develop' into feature/849-UpdateMap-new

Commit:
Добавлены интерфейсы для карт

Добавлен интерфейс менеджер  карт

Commit:
Исправлены ошибки при иницилизации карт, добавлены компонент search

Commit:
Незначительные изменения структуры кода( закоментировал app. module.ts строка 219) происходила двойная иницилизация компонентов

inyutin-maxim commented on sandbox/src/app/docs/maps/class/baidu/class-config.ts:
```diff
@@ -0,0 +1,313 @@
+            if (type == 'stop') {
+            }
+
+            if (type === 'circle') {
```
Посмотри на паттерн стратегия. ну и вообще лучше такие вещи выносить в отдельную приватную ф-ю хотя бы

Commit:
Доработка map, добавлен функционал для переключения карт, реализован метод Destoy (при выборе другой карты), работа с Leaftlet (настройка отображения карты),работа с Baidu, незначительные изменения структуры кода компонента

inyutin-maxim commented on sandbox/src/app/docs/maps/class/baidu/class-Initialize.ts:
```diff
@@ -53,13 +58,24 @@ export class Initialize implements InitMap {
     }
 
     destroyMap() {
+        let ds = document.getElementById('mapAPI');
```
вообще практика для Angular не очень, для работы с DOM деревом есть шаблонизатор, из кода ничего с DOM напрямую работать не должно

inyutin-maxim commented on sandbox/src/app/docs/maps/class/yandex/class-Initialize.ts:
```diff
@@ -39,24 +42,36 @@ export class Initialize implements InitMap {
             script.onerror = (error: any) => {
                 reject({ loaded: false, status: 'Error' });
             };
+            // let ds = document.getElementById('mapAPI');
```
закомментированный код нужно помечать как **todo**, с описанием почему было закомментировано, как и когда будет исправлено

inyutin-maxim commented on sandbox/src/app/docs/maps/class/leaflet/class-Initialize.ts:
```diff
@@ -32,6 +34,10 @@ export class Initialize implements InitMap {
             script.onerror = (error: any) => {
                 reject({ loaded: false, status: 'Error' });
             };
+            let ds = document.getElementById('mapAPI');
```
Вот этот кусок кучу раз дублируется нужно вынести в отдельную ф-ю

inyutin-maxim reviewed (changes requested):
Исправить замечания, поискать ангуляр обертки над картами, не использовать работу с DOM из кода, для этого есть шаблонизатор

Commit:
Доработка переключения карт, доработа уничтожение карты, добавлен Interface SearchPlacesMap

Commit:
Добавлены interfce i-inner:google, yandex, baidu, leflet (обертки для всех карт), добавлены markercluster для baidu и leaflet (загрузка point при иницилизации карты)

Commit:
Добавлена динамическая загрузка маркеров на карту ( в зависимости  от видимости карты),  исправлена иницилизация карты

Commit:
Изменения в структуре кода компонента, исправлены названия файлов, изменения отображения markerCluster на всех картах, работа с draw area yandex

inyutin-maxim commented on sandbox/src/app/docs/maps/class/baidu/class-Initialize.ts:
```diff
@@ -1,15 +1,16 @@
+@Injectable()
+export class InitializeBaidu implements IInitMap {
+    source: ILoadApiMap;
+    public loadScriptMap(source: ILoadApiMap): Promise<any> {
```
Вместо **Promise** нужно использовать **Observable**

inyutin-maxim commented on sandbox/src/app/docs/maps/class/baidu/class-Initialize.ts:
```diff
@@ -1,15 +1,16 @@
-    public loadScriptMap(source: LoadApiMap): Promise<any> {
+@Injectable()
+export class InitializeBaidu implements IInitMap {
+    source: ILoadApiMap;
```
Можно не создавать отдельно поле, вместо этого в конструкторе нужно добавлять ключевое слово **private**
```js
public loadScriptMap(private source: ILoadApiMap)
```

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-event.ts:
```diff
@@ -1,14 +1,15 @@
+@Injectable()
+export class EventsGoogle implements IEventMap {
 
     listenEvent<E>(objMap: any, eventName: string): Observable<E> {
```
Оно конечно хитро и наверняка работает, а что за тип **E**?

inyutin-maxim commented on sandbox/src/app/docs/maps/class/yandex/class-Initialize.ts:
```diff
@@ -1,14 +1,15 @@
+@Injectable()
+export class InitializeYandex implements IInitMap {
+    source: ILoadApiMap;
+    public loadScriptMap(source: ILoadApiMap): Promise<any> {
```
Можно не создавать отдельно поле, вместо этого в конструкторе нужно добавлять ключевое слово **private**
```js
public loadScriptMap(private source: ILoadApiMap)
```
Вместо **Promise** нужно использовать **Observable**

inyutin-maxim commented on sandbox/src/app/docs/maps/map-container/map-container.component.css:
```diff
@@ -7,3 +7,90 @@
+    background-color: rgba(181, 226, 140, 0.6);
+}
+
+    .marker-cluster-small div {
```
Отступы мне кажется здесь не нужны

Commit:
Исправленно отображение marker cluster leaflet, baiduб yandex, добавлены event dragend, zoomend,изменения структуры кода компонента (использование namespace)

Commit:
Прошелся по замечаниям, исправил DI, исправил различные не точности, ошибки Измененил названия у классов для более читабельности (при применении namescpace,) исправил только у google, исправил динамическую загрузку точек у baidu, yandex, leaflet

Commit:
Добавлены поля From и To, реализован поиск по карте google (вывод material autocomplete)

inyutin-maxim commented on sandbox/src/app/docs/maps/map-search/map-search.component.ts:
```diff
@@ -1,18 +1,45 @@
   }
 
+  stateCtrl = new FormControl();
+  SearchPoint: Observable<any[]>;
```
Необходимо как можно меньше использовать тип **any** в коде. Он добавлен в язык для обратной совместимости с **JS**. Строгая типизация позволит избежать лишних бессмысленных проверок на уровне компилятора

Commit:
Доработка поиска на карте (отображение marker на карте), построение маршрута (информация о маршруте)

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/baidu/class-config.ts:
```diff
@@ -0,0 +1,204 @@
+                for (let i = 0; i < mark.default.length; i++) {
+                    let item = mark.default[i];
+                    var icon = new BMap.Icon('./assets/icons_map/icon_hotel.png', new BMap.Size(60, 60));
+                    var marker = new BMap.Marker(new BMap.Point(item.Address.Lng, item.Address.Lat), {
```
Зачем создавать объект если он может не использоваться?

inyutin-maxim replied:
вот из за этого условия половина метода вообще может не иметь смысла

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/baidu/class-config.ts:
```diff
@@ -0,0 +1,204 @@
+        }
+
+        transitLayer(transit: any, boolean: boolean) {
+            if (boolean) {
```
в данном случае можно написать элегантнее и короче:
```js
transit.setMap(boolean ? this.objMap.map : null);
```
а во вторых я за такое именование буду очень ругаться, что это вообще значит переменная boolean?

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/baidu/class-config.ts:
```diff
@@ -0,0 +1,204 @@
+            }
+        }
+        trafficLayer(traffic: any, boolean: boolean) {
+            if (boolean) {
```
смотри коммент выше

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/baidu/class-config.ts:
```diff
@@ -0,0 +1,204 @@
+
+        }
+
+        draggableMap(boolean: any) {
```
Никогда не именуй так параметры или переменные

inyutin-maxim replied:
и какого типа твой булеан?

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/google/class-config.ts:
```diff
@@ -0,0 +1,347 @@
+
+                let func = (item, x1, y1) => {
+                    let b = this.inclusionMarkersPolygon(item, x1, y1);
+                    if (b === false) {
```
```js
if (!b)
```

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/google/class-config.ts:
```diff
@@ -0,0 +1,347 @@
+                }
+
+                let func = (item, x1, y1) => {
+                    let b = this.inclusionMarkersPolygon(item, x1, y1);
```
Необходимо давать более понятные имена переменным

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/google/class-event.ts:
```diff
@@ -0,0 +1,53 @@
+                if (event.placeId) {
+                    event.stop();
+                }
+                else {
```
не нужно

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/leaflet/class-Initialize.ts:
```diff
@@ -0,0 +1,64 @@
+
+        destroyMap(): void {
+            try {
+                let ds = document.getElementById('mapAPI');
```
Необходимо давать более понятные имена переменным

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/leaflet/class-config.ts:
```diff
@@ -0,0 +1,196 @@
+        }
+
+        transitLayer(transit: any, boolean: boolean) {
+            if (boolean) {
```
Смотри выше

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/leaflet/class-config.ts:
```diff
@@ -0,0 +1,196 @@
+            }
+        }
+        trafficLayer(traffic: any, boolean: boolean) {
+            if (boolean) {
```
Смотри выше

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/class/leaflet/class-config.ts:
```diff
@@ -0,0 +1,196 @@
+        draggableMap(boolean: any) {
+            if (boolean) {
+                console.log('grable false')
+                this.objMap.map.setOptions({
```
смотри выше
вообще нужно вынести весь одинаковый код в абстрактный класс

inyutin-maxim commented on sandbox/src/app/docs/layout/maps/map-selector/map-selector.component.ts:
```diff
@@ -0,0 +1,51 @@
+
+  private mapList(nameMap: string): mapInfoForSelect[] {
+    let temp: mapInfoForSelect[] = Object.keys(MapType)
+      .filter((type) => isNaN(<any>type))
```
Если я ничего не путаю в выборку попадет только значения NAN а для чего это?

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -204,46 +211,49 @@ export namespace Google {
+                        });
+                        marker["point"] = item;
+                        if (sending) {
+                            if (bounds.contains(marker.getPosition()) === true) {
```
Мне кажется метод **contains** возвращает **bool**, не нужно тогда писать `=== true`

inyutin-maxim commented on sandbox/src/app/docs/maps/class/leaflet/class-Initialize.ts:
```diff
@@ -3,65 +3,62 @@ import * as mapstyle from "../../class/google/maps.style.json";
+
+        destroyMap(): void {
+            try {
+                let ds = document.getElementById('mapAPI');
```
Необходимо давать более понятные имена переменным

inyutin-maxim commented on sandbox/src/app/docs/maps/class/yandex/class-config.ts:
```diff
@@ -15,7 +15,7 @@ export namespace Map.Yandex {
         }
 
-        showMarker(obj: any, markercluster: any) {
+        showMarker(obj: any) {
```
Необходимо как можно меньше использовать тип **any** в коде. Он добавлен в язык для обратной совместимости с **JS**. Строгая типизация позволит избежать лишних бессмысленных проверок на уровне компилятора

inyutin-maxim commented on sandbox/src/app/docs/maps/map-search/map-search.component.ts:
```diff
@@ -19,27 +24,56 @@ export class MapSearchComponent implements OnInit {
+
+  private SelectAutocompleteFrom(value: any) {
+    
+    //this.manager.getActiveMap().search.GetDetailsPointAutocomplete(value);
```
Закомментированный код необходимо удалять, или помечать как **TODO** с объяснением как и когда будет исправлено

Commit:
Рефакторинг компонента map (добавлены абстрактные классы,)

inyutin-maxim commented on sandbox/src/app/docs/maps/class/baidu/class-config.ts:
```diff
@@ -76,42 +76,42 @@ export namespace Map.Baidu {
-                    this.clearMap();
-                }
+            }
+            if (zoom == 3) {
```
Странный порядок

inyutin-maxim commented on sandbox/src/app/docs/maps/class/baidu/class-config.ts:
```diff
@@ -76,42 +76,42 @@ export namespace Map.Baidu {
+                    icon: icon,
+                    title: item.Hotelname,
+                });
+                if (sending) {
```
Я уже писал, что если условие не истино то создание объектов выше бессмысленно

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -81,9 +83,14 @@ export namespace Google {
                 let option: any;
+
+                // type 'stop'
                 if (type == 'stop') {
```
Просится в **switch**, строковые литералы в **enum**

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -166,8 +178,10 @@ export namespace Google {
                 let func = (item, x1, y1) => {
                     let b = this.inclusionMarkersPolygon(item, x1, y1);
+                    //remove all that poly not contains
                     if (b === false) {
```
```js
if (!b) {...}
```

inyutin-maxim replied:
Так же, перестань экономить на символах, пиши нормальные имена переменным

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -236,7 +251,7 @@ export namespace Google {
                         marker["point"] = item;
                         if (sending) {
-                            if (bounds.contains(marker.getPosition()) === true) {
+                            if (bounds.contains(marker.getPosition())) {
```
смерджить с условием выше

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -377,29 +389,20 @@ export namespace Google {
         }
 
         draggableMap(boolean: any) {
             if (boolean) {
```
Переименуй уже по человечески эту переменную

inyutin-maxim commented on sandbox/src/app/docs/maps/class/leaflet/class-config.ts:
```diff
@@ -155,29 +155,18 @@ export namespace Map.Leaflet {
         }
 
         draggableMap(boolean: any) {
             if (boolean) {
```
👿

inyutin-maxim commented on sandbox/src/app/docs/maps/map-search/map-search.component.ts:
```diff
@@ -20,60 +17,44 @@ export class MapSearchComponent  {
-      this.states = result;
+      this.ListAutocomplete = result;
     }
     if (value.length == 0) {
```
Странный порядок условий

inyutin-maxim commented on sandbox/src/app/docs/maps/map-search/map-search.component.ts:
```diff
@@ -20,60 +17,44 @@ export class MapSearchComponent  {
-      this.states = result;
+      this.ListAutocomplete = result;
     }
     if (value.length == 0) {
```
Странный порядок условий

inyutin-maxim commented on sandbox/src/app/docs/maps/map-selector/map-selector.component.ts:
```diff
@@ -1,20 +1,21 @@
 
-  constructor(private manager: Manager.Map.Manager) {
+  constructor(private manager: MapManager) {
     this.mapInfo = this.mapList('google');
```
Строковые литералы в **enum**

inyutin-maxim commented on sandbox/src/app/docs/maps/map-toolbar/map-toolbar.component.ts:
```diff
@@ -13,22 +13,30 @@ export class MapToolbarComponent implements OnInit {
   }
 
-  public zoomLevel(type:string) {
+  private zoomLevel(type:string) {
```
В чем причина становления методов приватными?

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -0,0 +1,277 @@
+    routeBuilder: AbstractRouteBuilder;
+
+    buildRoute(from: IPoint, to: IPoint, show?: boolean): void {
+        if (show) {
```
Инвертировать условие

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -0,0 +1,277 @@
+                    this.clearMap();
+                }
+
+                for (let marker of this.geo.markers) {
```
Условие должно быть над фором, не имеет смысла итерировать что то, если оно никуда не попадет
```js
if (markersVisible) {
```

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -0,0 +1,277 @@
+
+        let radiansDifferenceOfLongitudes = radiansPointLongitude - radiansCircleCenterLongitude;
+
+        let arcTanganceOfTop = Math.sqrt(Math.pow(Math.cos(radiansPointLatitude) * Math.sin(radiansDifferenceOfLongitudes), 2) + Math.pow(Math.cos(radiansCircleCenterLatitude) * Math.sin(radiansPointLatitude) - Math.sin(radiansCircleCenterLatitude) * Math.cos(radiansPointLatitude) * Math.cos(radiansDifferenceOfLongitudes), 2));
```
Вынести формулы в приватные ф-ции

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -0,0 +1,277 @@
+            this.marker.point = marker.point;
+
+            if (this.markerCluster) {
+                this.markerCluster.addMarker(marker, true);
```
Если условие ложно то все происходящее выше в методе бессмысленно

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -0,0 +1,277 @@
+        }
+    }
+
+    toggleMapDragging(enabled?: boolean) {
```
С чем связано нуллабельность?

inyutin-maxim commented on sandbox/src/app/docs/maps/new/entity/geo-container.ts:
```diff
@@ -0,0 +1,40 @@
+    clusters: Array<AbstractMarkerCluster>;
+
+    private clearArray(array: Array<any>) {
+        while(array.length)
```
Обернуть фигурными скобками, а вообще капец какой странный метод очистить массив

inyutin-maxim commented on sandbox/src/app/docs/maps/new/providers/google/map.ts:
```diff
@@ -0,0 +1,85 @@
+
+            if (apiScript.readyState) {
+                apiScript.onreadystatechange = () => {
+                    if (apiScript.readyState === "loaded" || apiScript.readyState === "complete") {
```
Строковые литералы в enum

Commit:
Рефакторинг map (Исправлена иницилизация компонента, изменения abstract class,  иницилизация events карты, изменения структуры кода, исправлены ошибки

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -13,27 +13,27 @@ export namespace Google {
                     clickable: true,
                     icon: { url: './assets/icons_map/icon_hotel.png' },
-                    title: ''
+                    title: point.Hotelname
```
**point.HotelName**

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -64,10 +64,9 @@ export namespace Google {
+                let option;
                 let drawingManager;
-                var radius: number = 10000;
+                let radius: number = 10000;
```
**let** используется в сценариях когда переменная предполагает изменение значения, иначе нужно использовать **const**

inyutin-maxim commented on sandbox/src/app/docs/maps/class/google/class-config.ts:
```diff
@@ -82,106 +81,93 @@ export namespace Google {
-                            map: map,
-                            clickable: false, strokeColor: '#1E90FF',
+                switch (type) {
+                    case 'stop':
```
Строковые литералы в **enum**

inyutin-maxim commented on sandbox/src/app/docs/maps/class/leaflet/class-config.ts:
```diff
@@ -82,20 +82,19 @@ export namespace Map.Leaflet {
+            if (zoom < 3) {
                 this.clearMap();
             }
+            if (sending) {
```
Исправь форматирование текста, условие и цикл на одном уровне

Commit:
Востановлены: события карты (idle) изменения границы области видимости карты
события клика по карте (получение координат или placeId для дальнейшей работы с этими даными)
Востановлена работа transitLayer, trafficLayer (отображение пробок и общественного транспорта на карте)
Востановлены методы изменения zoom (ZoomIn, ZoomOut), незначительные изменения структуры кода

inyutin-maxim commented on sandbox/src/app/docs/maps/map-toolbar/map-toolbar.component.ts:
```diff
@@ -1,43 +1,39 @@
-  private zoomLevel(type:string) {
-    this.manager.getActiveMap().config.setZoomLevel(type)
+  private zoomLevel(type: string) {
+    if(type === 'plus'){
```
Все строковые литералы необходимо выносить в **enum**

inyutin-maxim commented on sandbox/src/app/docs/maps/map-toolbar/map-toolbar.component.ts:
```diff
@@ -1,43 +1,39 @@
-  private onChangeTransit(event:any){
-    this.manager.getActiveMap().config.transitLayer(event.checked)
+  private onChangeTransit(event: any) {
+    this.manager.map.config.toggleTransitLayer(event.checked);
```
поле **map** нужно сделать приватным и к нему не должно быть доступа снаружи
```JS
this.manager.map
```

Так делать не нужно, как раньше было правильнее
```js
this.manager.getActiveMap()
```

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -222,13 +224,13 @@ export abstract class AbstractConfig implements IConfig {
-        let currentZoom = this.map.options.zoom;
-        this.map.setZoom(currentZoom + 1);
+        let currentZoom = this.map.api.getZoom();
+        this.map.api.setZoom(currentZoom + 1);
```
В абстракциях **НЕЛЬЗЯ** использовать конкретные методы реализации карты

Commit:
Доработка events (использование Observable) изменения структуры кода перенос некоторых методов из абстракного класса в конкретный, прошелся по замечаниям

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -246,6 +230,9 @@ export abstract class AbstractConfig implements IConfig, IInitMap {
                 visible: true
             });
             this.marker.point = marker.point;
+            marker = new google.maps.Marker(marker);
```
В абстрактном классе не должно быть вызовов конкретных карт

inyutin-maxim commented on sandbox/src/app/docs/maps/new/entity/map-manager.ts:
```diff
@@ -36,24 +37,28 @@ export class MapManager {
+                    this.currentMap.events.idle<void>().subscribe(() => {
+                        this.currentMap.config.drawMarkersOnMap();
+                    });
+                    this.currentMap.events.mapClicked<IEventClikMap>().subscribe((IEventClikMap) => {
```
Опечатка **IEventClikMap** должно быть **IEventCliсkMap**

Commit:
Работа с markercluster, изменения структуры кода, добавлены новые  методы GetGatailsPoint() и GetAdress() получение детальной информации о точке по GooglePlaceID при клике по карте и получение информации о точке по координатам

inyutin-maxim commented on sandbox/src/app/docs/maps/map-selector/map-selector.component.ts:
```diff
@@ -14,10 +14,11 @@ export class MapSelectorComponent {
 
   constructor(private manager: MapManager) {
-    this.mapInfo = this.mapList('google');
+    this.mapInfo = this.mapList('GOOGLE');
```
Все строковые литералы необходимо выносить в **enum**

inyutin-maxim commented on sandbox/src/app/docs/maps/new/interfaces/i-position.ts:
```diff
@@ -1,4 +1,5 @@
-export interface IPosition {
+export class IPosition {
```
А что у тебя класс делает в папке с интерфейсами

Commit:
Изменения метода drawMarkersOnMap()  изменения кода в абстракном классе и в конкретной реализации, добавлен метод boundsContainsMarker() вхождение точки в границы  видимой карты

Commit:
Незначительные изменения метода  drawMarkersOnMap()

inyutin-maxim commented on sandbox/src/app/docs/maps/map-toolbar/map-toolbar.component.html:
```diff
@@ -1,7 +1,7 @@
 <div class="buttons-row">
-  <button mat-button (click)="zoomLevel('plus')" color="primary">Zoom Level +</button>
-  <button mat-button (click)="zoomLevel('minus')" color="primary">Zoom Level-</button>
+  <button mat-button (click)="zoomLevel('In')" color="primary">Zoom Level +</button>
```
Ну Андрюха, ну энамы же

inyutin-maxim commented:
И разреши конфликты пожалуйста

ALNovak committed:
Merge branch 'develop' into feature/849-UpdateMap-new

Commit:
Изменения структуры кода компонента, Добавлены методы удаления гeoобъектов, добавлены новые методы MarkerCluster, исправлены незначительные ошибки

Commit:
Изменения структуры кода компонента, Добавлены методы удаления гeoобъектов, добавлены новые методы MarkerCluster, исправлены незначительные ошибки

inyutin-maxim commented on sandbox/src/app/app.module.ts:
```diff
@@ -141,11 +141,15 @@ import { GoogleRouteBuilder } from './docs/maps/new/providers/google/route';
 import { GoogleMarker } from './docs/maps/new/providers/google/marker';
 import { GoogleMapOptions } from './docs/maps/new/providers/google/entity/google-map-options';
 import { AbstractMap } from './docs/maps/new/abstract/abstract-map'
+<<<<<<< HEAD
```
зачем код с конфликтами коммитить?

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -54,32 +56,69 @@ export abstract class AbstractConfig implements IConfig, IInitMap {
+
+                this.clearAllMap();
             }
             if (zoom > 5) {
```
А при зуме равном 4 и 5 что должно произойти?

inyutin-maxim commented on sandbox/src/app/docs/maps/new/providers/google/config.ts:
```diff
@@ -25,9 +27,89 @@ let trafficLayer;
+        if (this.map.geo.markers.length > 1) {         
+                 
+        }
+        if (this.map.geo.markers.length == 1) {
```
для чего нужны пустые условия?

inyutin-maxim reviewed (changes requested):
В целом код становится все чище и лучше, но проблем пока хватает, коммитить конфликты никуда не годится

ALNovak committed:
Merge branch 'develop' into feature/849-UpdateMap-new

Commit:
Изменения структуры кода компонента, Добавлены методы удаления гeoобъектов, добавлены новые методы MarkerCluster, исправлены незначительные ошибки

Commit:
Изменения структуры кода компонента, Добавлены методы удаления гeoобъектов, добавлены новые методы MarkerCluster, исправлены незначительные ошибки

Commit:
Изменения структуры кода компонента, Добавлены методы  работы с картой,, исправлены незначительные ошибки

Commit:
Изменены методы drawArea(), drawCircle(), изменения структуры кода, исправлены незначительные ошибки

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -82,17 +82,17 @@ export abstract class AbstractConfig implements IConfig, IInitMap {
                 this.clearAllMap();
+
             }
             if (this.getZoom() > 5) {
```
Писал ранее, спрошу еще раз, а что случится на масштабе 4 или 5?

inyutin-maxim commented on sandbox/src/app/docs/maps/new/interfaces/i-polygon.ts:
```diff
@@ -1,7 +1,7 @@
 
 export interface IPolygonOptions {
-    path: Array<IPoint>;
+    path: Array<any>;
```
А что случилось что возникла необходимость в **ANY**?

inyutin-maxim commented on sandbox/src/app/docs/maps/new/providers/google/config.ts:
```diff
@@ -66,10 +70,21 @@ export class GoogleConfig extends AbstractConfig {
 
+        googlemarker["geoPoint"] = marker;
+
+        google.maps.event.addListener(googlemarker, 'click', (event) => {
```
события тоже в enum 'click'

Commit:
Реализованы методы getAddress, Getdetails, исправлены не значительные ошибки

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -230,8 +243,8 @@ export abstract class AbstractConfig implements IConfig, IInitMap {
 
-        let markerLatitude = marker.point.latitude;
-        let markerLongitude = marker.point.longitude;
+        let markerLatitude = marker.point.position.latitude;
```
дублирование кода, вынести в приватную ф-ю

inyutin-maxim commented on sandbox/src/app/docs/maps/new/abstract/abstract-config.ts:
```diff
@@ -265,8 +278,8 @@ export abstract class AbstractConfig implements IConfig, IInitMap {
 
-        let pointLatitude = marker.point.latitude;
-        let pointLongitude = marker.point.longitude;
+        let pointLatitude = marker.point.position.latitude;
```
см выше

inyutin-maxim commented on sandbox/src/app/docs/maps/new/providers/google/classes/address-settings.ts:
```diff
@@ -0,0 +1,11 @@
+    street_number: string;
+    route: string;
+    locality: string;
+    postal_town: string;
```
camelCase

Commit:
Доработка метода обратного геокодирования, и формирования аддреса, незначительные изменения в структуре кода, добавленны дополнительные поля IPoint

Commit:
Незначительные изменения структуры кода, доработка удаления markers, polygon и др. geo объектов

inyutin-maxim commented on sandbox/src/app/docs/maps/map-search/map-search.component.ts:
```diff
@@ -84,23 +84,19 @@ export class MapSearchComponent {
-    this.manager.getActiveMap().config.loadMarkers = false;
+    this.manager.getActiveMap().config.getDetailsPoint(placeid).subscribe(point => {
+      this.manager.getActiveMap().config.showMarker(point);
+    
```
Место для рекламы?

inyutin-maxim reviewed (changes requested):
Форматируй пожалуйста документы

Commit:
Незначительные изменения в структуре кода компонента

Commit:
Незначительные изменения структуры кода компонента, изменения Address component

Commit:
Незначительные изменения в структуре кода AddressComponent (формирование адреса для  point)

Commit:
Изменения в структуре кода, изменения в Addresscomponent config , добавлены базовые классы Base Polygon, Base Polyline, Base Cicle

Commit:
Незначительные изменения в структуре кода, изменения Address component, добавлены базовые классы Base Cicle, Base Polyline, Base Polygon

Commit:
Изменеия структуры кода, для примера добавлен новый провайдер (необходимы определенные настройки)

Commit:
Изменения в структуре кода компонента: Исправлена ошибка подключекния провайдера baidu

Commit:
Изменения в  струтуре кода компонента , создан метод getBaseMarker(), изменения  базовых классов

Commit:
Изменения структуры  кода, добавлен метод getDistanse() , изменения метода showRoute()

Commit:
Изменения в структуре кода компонента, добавлен провайдер yandex (основа файлов и иницилизация, доработки по получении информации о маршруте)

Commit:
Изменения в структуре компонента,  реализация отображения маршрута маршрута по типу: CAR, FLY, исправлены незначительные ошибки

Commit:
Изменения в структуре кода компонента , buildRoute, map-search component

Commit:
Доработка построения маршрута

Commit:
Незначительные изменения в структуре компонента, исправление отображения построение маршрута

Commit:
Незначительные изменения в структуре компонента, работа с  emit

Commit:
Незначительные изменения в структуре компонента( изменени поиска по карте)

Commit:
Изменения структуры кода компонента, исправлены незначительные ошибки,

Commit:
Cоздание и настройка модуля map, незначительные изменения в структуре кода

Commit:
Доработан модуль maps, исправлены не значительные ошибки, изменения структуры модуля
  Добавлены Callback событий карты (радиуса, маркера)для взаимодействия с картой

Commit:
Доработка модуля (добавлены основные callback) использованы Subject, незначительное изменения структуры кода

Commit:
Изменения структуры кода, работа с Callback

Commit:
Доработки модуля, исправлены незначительные ошибки

Commit:
Изменения структуры модуля компонента, добавлен провайдер leaflet (исправлены незначительные ошибки), описание методов

Commit:
Незначительные изменения в структуре кода, добавлен метод getSelectedMarker()

Commit:
Рефакторинг модуля map, изменения документации в коде, изменения в point, генерация JSON point

Commit:
Незначительные изменения в коде модуля
